### PR TITLE
add man page

### PR DESCRIPTION
**Pull request by tillschaefer**
see https://bugs.gentoo.org/676928

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,6 +3,8 @@ PREFIX = /usr/local
 SYSCONFDIR = /etc
 # used for the default storage path
 SHAREDSTATEDIR = /var
+# used for the default manpage path
+MANPAGEDIR = $(PREFIX)/share/man
```
Comment (tillschaefer) [inline Makefile]: I am unsure if the PREFIX is used correctly here

File: repository.eselect.5.in [hunk 2]
```diff
@@ -0,0 +1,169 @@
+.\" -*- coding: utf-8 -*-
+.\" Copyright 2019 Gentoo Authors
+.\" Distributed under the terms of the GNU General Public License v2
+.\"
```
Comment (tillschaefer) [inline repository.eselect.5.in:4]: Is the GPL2 header ok?

File: repository.eselect.5.in [hunk 3]
```diff
@@ -0,0 +1,169 @@
+.\" -*- coding: utf-8 -*-
+.\" Copyright 2019 Gentoo Authors
```
Comment (mgorny) [inline repository.eselect.5.in]: Put your own name here, 'Gentoo Authors' is not correct for entirely external projects.

File: repository.eselect.5.in [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+.\" Distributed under the terms of the GNU General Public License v2
```
Comment (mgorny) [inline repository.eselect.5.in]: eselect-repo is BSD-2, so GPL2 is a no-go here.

File: repository.eselect.5.in [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+.\"
+.TH repository.eselect 5 "31 Januar 2019" "Gentoo Linux" "eselect"
```
Comment (mgorny) [inline repository.eselect.5.in]: January

**Review by mgorny — CHANGES_REQUESTED** [review]
I've left a few initial comments. However, the further I go, my eyes go all crazy over the completely unreadable semantics. Could you use a readable markup like ReST instead? I think it comes with a nice manpage output format; alternatively you could use Sphinx if pure ReST is insufficient.

**Comment by tillschaefer** [comment]
> I've left a few initial comments. However, the further I go, my eyes go all crazy over the completely unreadable semantics. Could you use a readable markup like ReST instead? I think it comes with a nice manpage output format; alternatively you could use Sphinx if pure ReST is insufficient.

It will cost me some time to get used to the converter and to rewrite the man page. If you say it is a must, I will do so, but I would like to avoid it if possible and user my time rosources on something else :-)

File: Makefile (hunk 1 above)
Comment (mgorny) [reply Makefile]: Please name it `MANDIR` to match what autoconf uses.

File: Makefile [hunk 6]
```diff
@@ -30,15 +32,23 @@ repository.eselect: repository.eselect.in Makefile
...
 
-install: repository.eselect
+install: repository.eselect repository.eselect.5
 	install -d $(DESTDIR)$(HELPERDIR)
 	install -m0755 eselect-repo-helper $(DESTDIR)$(HELPER)
 	install -d $(DESTDIR)$(ESELECTDIR)
 	install -m0644 repository.eselect $(DESTDIR)$(ESELECTDIR)/
 	install -d $(DESTDIR)$(CONFIGDIR)
 	install -m0644 eselect-repo.conf $(DESTDIR)$(CONFIG)
+	install -d $(DESTDIR)$(MANPAGEDIR)/man5/
```
Comment (mgorny) [inline Makefile]: Please remove the trailing slash here.

File: repository.eselect.5.in [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+.fam C
+\fBeselect\fP \fBrepository\fP [\fBhelp\fP|\fBusage\fP|\fBversion\fP]
+\fBeselect\fP \fBrepository\fP \fBadd\fP \fIname\fP \fIsync-type\fP \fIsync-uri\fP
+\fBeselect\fP \fBrepository\fP \fBdisable\fP [\fB-f\fP] (\fIname\fP|\fIindex\fP)
+\fBeselect\fP \fBrepository\fP \fBenable\fP (\fIname\fP|\fIindex\fP)
+\fBeselect\fP \fBrepository\fP \fBlist\fP [\fB-i\fP]
+\fBeselect\fP \fBrepository\fP \fBremove\fP [\fB-f\fP] (\fIname\fP|\fIindex\fP)
+
+.SH DESCRIPTION
+\fBrepository.eselect\fP is an eselect module for manipulating /etc/portage/repos.conf entries to third party repositories (also known as overlays). This supersedes the need for app-admin/layman to list, configure, and handle synchronizations of alternate repositories except for those which the package manager does not (eg. mercurial, bazaar and g-sorcery in portage).
```
Comment (mgorny) [inline repository.eselect.5.in]: Could you wrap at 72 chars? Also it controls entries to *all* repositories, including ::gentoo. Also, I don't think listing layman is exactly the way to describe eselect-repo.

File: Makefile [hunk 8]
```diff
@@ -30,15 +32,23 @@ repository.eselect: repository.eselect.in Makefile
...
+	rm -f $@.tmp
+	sed 's=/etc/eselect/repository.conf=$(CONFIG)=' $< > $@.tmp
+	chmod a+r $@.tmp
+	mv $@.tmp $@
+
 clean:
 	rm -f repository.eselect
 
-install: repository.eselect
+install: repository.eselect repository.eselect.5
```
Comment (mgorny) [inline Makefile:44]: Also add it to `all` target.

File: repository.eselect.5.in [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+.\" Distributed under the terms of the 2-clause BSD license
+.\"
+.TH repository.eselect 5 "31 January 2019" "Gentoo Linux" "eselect"
+.SH NAME
+\fBrepository.eselect\fP - Manage repos.conf via eselect
+.SH SYNOPSIS
+.nf
+.fam C
+\fBeselect\fP \fBrepository\fP [\fBhelp\fP|\fBusage\fP|\fBversion\fP]
+\fBeselect\fP \fBrepository\fP \fBadd\fP \fIname\fP \fIsync-type\fP \fIsync-uri\fP
```
Comment (mgorny) [inline repository.eselect.5.in]: I think it's customary to put user parameters in `<...>`.

File: repository.eselect.5.in [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+.SH ACTION: DISABLE
+Disable specified repositories without removing their contents
+.PP
+\fBeselect\fP \fBrepository\fP \fBdisable\fP [\fB-f\fP] (\fIname\fP|\fIindex\fP)
+.PP
+Parameters:
+.TP
+.B
+\fB-f\fP
+Force potentially dangerous removals. Required for unregistered repositories and those without sync attributes.
```
Comment (mgorny) [inline repository.eselect.5.in]: You could explain that dangerous = repositories that can't be readded via enable.

File: repository.eselect.5.in [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+.\"
+.TH repository.eselect 5 "31 January 2019" "Gentoo Linux" "eselect"
+.SH NAME
+\fBrepository.eselect\fP - Manage repos.conf via eselect
+.SH SYNOPSIS
+.nf
+.fam C
+\fBeselect\fP \fBrepository\fP [\fBhelp\fP|\fBusage\fP|\fBversion\fP]
+\fBeselect\fP \fBrepository\fP \fBadd\fP \fIname\fP \fIsync-type\fP \fIsync-uri\fP
+\fBeselect\fP \fBrepository\fP \fBdisable\fP [\fB-f\fP] (\fIname\fP|\fIindex\fP)
```
Comment (mgorny) [inline repository.eselect.5.in]: Also `...` when multiple values can be passed.

File: repository.eselect.5.in [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+.PP
+State suffixes:
+.TP
+.B
+*
+Repository is enabled
+.TP
+.B
+#
+Repositorys sync information is out of date and needs to be updated (via
```
Comment (mgorny) [inline repository.eselect.5.in]: Repository's

File: repository.eselect.5.in [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+disable/enable) or is customized
```
Comment (mgorny) [inline repository.eselect.5.in]: s/is/has been/

File: repository.eselect.5.in [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+.TP
+.B
+@
+Repository is not listed by name in the official, published list
```
Comment (mgorny) [inline repository.eselect.5.in]: s/official, published/remote/. Technically you can use any list you like.

Also, probably something like 'repository with a matching name is ...' would be more correct.

**Review by mgorny — CHANGES_REQUESTED** [review]
I'm sorry that this took so long.

File: repository.eselect.5.in (hunk 9 above)
Comment (tillschaefer) [reply repository.eselect.5.in]: AFAIK this is not true for man pages (they are just underlined there / gross: \fI). At least this is the was the man page of portage is written and the man pages of man itself. Thus, I think it is better to keep in line with the other documentations of related tools.

**Comment by tillschaefer** [comment]
I have applied you change requests (except the <...> request / see above) and have rewritten the description text to not reference layman anymore.

**Comment by heroin-moose** [comment]
Christ, this thing is ugly. How about a much cleaner [mandoc](https://mandoc.bsd.lv) version?

```
.Dd $Mdocdate: March 14 2019 $
.Dt REPOSITORY.ESELECT 8
.Os
.Sh NAME
.Nm repository.eselect
.Nd manage ebuild repositories via eselect
.Sh SYNOPSIS
.Nm eselect repository
.Op Cm help | usage | version
.Nm eselect repository
.Cm list Op Fl i
.Nm eselect repository
.Cm enable Ar index Ns | Ns Ar name Ar ...
.Nm eselect repository
.Cm disable Oo Fl f Oc Ar index Ns | Ns Ar name Ar ...
.Nm eselect repository
.Cm add Ar name Ar sync-type Ar sync-uri
.Nm eselect repository
.Cm remove Oo Fl f Oc Ar index Ns | Ns Ar name Ar ...
.Sh DESCRIPTION
.Nm
is an
.Xr eselect 1
module for managing ebuild repositories.
.Pp
It's main purpose is to manage Portage repositories confguration,
synchronization of the the repositories is handled by the package manager.
.Pp
In order to do it's job
.Nm
keeps track of the registered repositories by using the curated remote repository list.
On the first run this list is cached and is automatically updated by
.Nm
every now and then.
.Pp
It is still possible to manually add/remove repositories,
.Nm
will handle it just fine.
.Pp
The following commands are available:
.Pp
.Bl -tag -width Ds
.It Cm help | usage
Show usage and exit.
.It Cm version
Show
.Xr eselect 1
and
.Nm
versions and exit.
.It Cm list Op Fl i
List known repositories.
If
.Fl i
is given,
.Nm
will list only installed repositories.
.Pp
The output consists of three columns: repository index, repository name and an
optional repository state.
If the third column is empty then this repository is not enabled.
.Pp
The states are as follows:
.Bl -tag -width Ds
.It Sq *
Repository is enabled.
.It Sq #
Repository configuration does not match the configuration in the remote
repository list.
If you have installed this repository from the list, you need to update it's
configuration by disabling and enabling it again.
.It Sq @
Repository is not listed in the remote repository list.
.El 
.Pp
For example:
.Bd -literal -offset indent
# eselect repository list
Available repositories:
  [1] master #
  [2] extras *
  [3] mystuff @
  [4] some-dudes-repo
  [5] another-dudes-repo

# eselect repository list -i
Available repositories:
  [1] master #
  [2] extras *
  [3] mystuff @
.Ed
.It Cm enable Ar index Ns | Ns Ar name Ar ...
Enable the specified repositories from the list.
.Pp
For example:
.Bd -literal -offset indent
# eselect repository list | grep -E '(foo|bar)'
  [5] foo
  [6] bar 
# eselect repository enable 5 bar
.Ed
.It Cm disable Oo Fl f Oc Ar index Ns | Ns Ar name Ar ...
Disable the specified repositories without removing their contents.
If
.Fl f
is present,
.Nm
is allowed to disable unregistered repositories and repositories without
sync-uri.
.It Cm add Ar name Ar sync-type Ar sync-uri
Add a new repository named
.Ar name
where
.Ar sync-type
is a syncing method (e.g. git or rsync) and
.Ar sync-uri
is a URI to the repository source.
See
.Xr portage 5
for a list of valid
.Ar sync-type
values.
.Pp
For example, if you want to add some third-party repository via Git you should
type something like this:
.Bd -literal -offset indent
# eselect repository add foobar git https://git.example.com/foobar.git
.Ed
.Pp
Note that you probably need this command only if using
.Nm eselect repository Cm enable
is not possible.
.It Cm remove Oo Fl f Oc Ar index Ns | Ns Ar name Ar ...
Remove installed repositories along with their contents.
If
.Fl f
is present,
.Nm
is allowed to remove unregistered repositories and repositories without
sync-uri.
.El
.Sh FILES
.Bl -tag -width Ds
.It Pa /etc/eselect/repository.conf
Configuration file for
.Nm .
.It Pa /etc/portage/repos.conf/
Default repository configuration directory.
See
.Xr portage 5
for details.
.It Pa $HOME/.cache/repositories.xml
Default cache file for remote repository list.
.It Pa /var/db/repos/
Default root directory for repositories.
.El
.Sh SEE ALSO
.Xr eselect 1
.Xr portage 5
```

First draft, I guess.

Rendered:

```
REPOSITORY.ESELECT(8)  Gentoo System Manager's Manual  REPOSITORY.ESELECT(8)

NAME
     repository.eselect — manage ebuild repositories via eselect

SYNOPSIS
     eselect repository [help | usage | version]
     eselect repository list [-i]
     eselect repository enable index|name ...
     eselect repository disable [-f] index|name ...
     eselect repository add name sync-type sync-uri
     eselect repository remove [-f] index|name ...

DESCRIPTION
     repository.eselect is an eselect(1) module for managing ebuild reposi‐
     tories.

     It's main purpose is to manage Portage repositories confguration, syn‐
     chronization of the the repositories is handled by the package manager.

     In order to do it's job repository.eselect keeps track of the regis‐
     tered repositories by using the curated remote repository list.  On the
     first run this list is cached and is automatically updated by
     repository.eselect every now and then.

     It is still possible to manually add/remove repositories,
     repository.eselect will handle it just fine.

     The following commands are available:

     help | usage
             Show usage and exit.

     version
             Show eselect(1) and repository.eselect versions and exit.

     list [-i]
             List known repositories.  If -i is given, repository.eselect
             will list only installed repositories.

             The output consists of three columns: repository index, reposi‐
             tory name and an optional repository state.  If the third col‐
             umn is empty then this repository is not enabled.

             The states are as follows:

             ‘*’     Repository is enabled.

             ‘#’     Repository configuration does not match the configura‐
                     tion in the remote repository list.  If you have
                     installed this repository from the list, you need to
                     update it's configuration by disabling and enabling it
                     again.

             ‘@’     Repository is not listed in the remote repository list.

             For example:

                   # eselect repository list
                   Available repositories:
                     [1] master #
                     [2] extras *
                     [3] mystuff @
                     [4] some-dudes-repo
                     [5] another-dudes-repo

                   # eselect repository list -i
                   Available repositories:
                     [1] master #
                     [2] extras *
                     [3] mystuff @

     enable index|name ...
             Enable the specified repositories from the list.

             For example:

                   # eselect repository list | grep -E '(foo|bar)'
                     [5] foo
                     [6] bar
                   # eselect repository enable 5 bar

     disable [-f] index|name ...
             Disable the specified repositories without removing their con‐
             tents.  If -f is present, repository.eselect is allowed to dis‐
             able unregistered repositories and repositories without sync-
             uri.

     add name sync-type sync-uri
             Add a new repository named name where sync-type is a syncing
             method (e.g. git or rsync) and sync-uri is a URI to the reposi‐
             tory source.  See portage(5) for a list of valid sync-type val‐
             ues.

             For example, if you want to add some third-party repository via
             Git you should type something like this:

                   # eselect repository add foobar git https://git.example.com/foobar.git

             Note that you probably need this command only if using eselect
             repository enable is not possible.

     remove [-f] index|name ...
             Remove installed repositories along with their contents.  If -f
             is present, repository.eselect is allowed to remove unregis‐
             tered repositories and repositories without sync-uri.

FILES
     /etc/eselect/repository.conf
             Configuration file for repository.eselect.

     /etc/portage/repos.conf/
             Default repository configuration directory.  See portage(5) for
             details.

     $HOME/.cache/repositories.xml
             Default cache file for remote repository list.

     /var/db/repos/
             Default root directory for repositories.

SEE ALSO
     eselect(1) portage(5)

Gentoo/linux                   March 14, 2019                   Gentoo/linux

```

**Comment by tillschaefer** [comment]
That is not only a different format, but also a completely different content. While I like the layout and some of the changes, it also looses highlighting and ignores some of the previously discussed topics (such as the force removal text). It will also add the mandoc dependency (not stabilized yet in ::gentoo).

@mgorny: how should we proceed?

**Comment by heroin-moose** [comment]
@tillschaefer, it's not, groff handles mandoc very well (for example, see tmux, ntpd or sshd man pages), so sys-apps/man-db is enough. There is no problem with highlighting too (again, see the manual pages in your favourite pager).

**Comment by mgorny** [comment]
I'm not sure where we ended with this but eselect-repo has a manpage now, so closing.

**Pull request closed without merging**